gpbft/chain.go

```diff
@@ -48,6 +54,27 @@ type TipSet struct {
 	Commitments [32]byte
 }
 
+var EMPTY_COMMITMENTS = [32]byte{}
+
+func (ts *TipSet) Validate() error {
+	if len(ts.Key) == 0 {
+		return errors.New("tipset key must not be empty")
+	}
+	if len(ts.Key) > TIPSET_KEY_MAX_LEN {
+		return errors.New("tipset key too long")
+	}
+	if len(ts.PowerTable) == 0 {
+		return errors.New("power table CID must not be empty")
+	}
+	if len(ts.PowerTable) > CID_MAX_LEN {
+		return errors.New("power table CID too long")
+	}
+	if !bytes.Equal(ts.Commitments[:], EMPTY_COMMITMENTS[:]) {
```

Comment:
Doesn't really matter (we'll reject it when we compare it with our own chain) but I guess this is fine.

Reply:
Good point, I removed this check so it doesn't trip up later. 

This means that a message with non-empty commitments will be silently ignored after being received rather than being marked as invalid and not propagating.